tx.go

```diff
@@ -323,7 +357,7 @@ func (tx *Tx) canAddChange(available uint64, standardFees *Fee) bool {
 	changeOutputFee += uint64(changeP2pkhByteLen * standardFees.MiningFee.Satoshis / standardFees.MiningFee.Bytes)
 
 	// not enough change to add a whole change output so don't add anything and return
-	return available >= changeOutputFee
+	return available >= (changeOutputFee + DustLimit)
```

Comment:
Yeah that would be fine, I'd left this in thinking this could be a good "fail fast" way of ending computation early but you're right, this wouldn't work for adding to an existing output. I don't think there is any harm removing this and the tests should all still pass

Reply:
okay lemme try now